### ISSUE TITLE
Ability to remap pointer types

### ISSUE THREAD

**Issue by RandyPasion** [bottlenoselabs/c2cs#111]
Currently remapping is only evaluated on element types (e.g. remapping `someType` will apply the same remapping to `someType*`)

Allow defining remapping of pointer types separately from element type.

This can be particularly useful for safe pointer management in C#. A common use case would be to remap `someType*` to a class inheriting `SafeHandle` (e.g. `SomeTypeHandle : SafeHandle`) that the user defines for managing the pointer.

Secondary improvement here - prioritize user remappings to defaults. Some types are default remapped ((`char*` to `CString` for example). If a user defines a different remapping from this, it should prefer the user's remapping.

**Comment by lithiumtoast** [comment]
Makes sense; sounds like this would go well with the idea of trying to work on right now: C# plugin. Use a C# library to control generating C# code rather than a config file. This would help with automating custom wrappers as custom C# code can be written to help transform the bindings into a desired wrapper.

**Comment by lithiumtoast** [comment]
I found a way forward to implement this. It requires some major refactoring however... I'll be working on it when I can.

**Comment by lithiumtoast** [comment]
Hey @RandyPasion, sorry for the delay in this.

> A common use case would be to remap someType* to a class inheriting SafeHandle (e.g. SomeTypeHandle : SafeHandle) that the user defines for managing the pointer.
The part about pointers should be already fixed. How it works is that the type name needs to be an exact match for remapping. Pointer types have a more complete name (longer) and thus should be remapped (short-circuit the logic) before checking the element type for remapping.

> Secondary improvement here - prioritize user remappings to defaults. 
I added support for this in #169.

Thanks!
